xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.cpp

```diff
@@ -33,50 +33,121 @@ CDebugRenderer::~CDebugRenderer()
   }
 }
 
-void CDebugRenderer::SetInfo(std::string &info1, std::string &info2, std::string &info3, std::string &info4)
+void CDebugRenderer::SetInfo(DEBUG_INFO_PLAYER& info)
 {
   m_overlayRenderer.Release(0);
 
-  if (info1 != m_strDebug[0])
+  if (info.audio != m_strDebug[0])
   {
-    m_strDebug[0] = info1;
+    m_strDebug[0] = info.audio;
     if (m_overlay[0])
       m_overlay[0]->Release();
     m_overlay[0] = new CDVDOverlayText();
     m_overlay[0]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[0]));
   }
-  if (info2 != m_strDebug[1])
+  if (info.video != m_strDebug[1])
   {
-    m_strDebug[1] = info2;
+    m_strDebug[1] = info.video;
     if (m_overlay[1])
       m_overlay[1]->Release();
     m_overlay[1] = new CDVDOverlayText();
     m_overlay[1]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[1]));
   }
-  if (info3 != m_strDebug[2])
+  if (info.player != m_strDebug[2])
   {
-    m_strDebug[2] = info3;
+    m_strDebug[2] = info.player;
     if (m_overlay[2])
       m_overlay[2]->Release();
     m_overlay[2] = new CDVDOverlayText();
     m_overlay[2]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[2]));
   }
-  if (info4 != m_strDebug[3])
+  if (info.vsync != m_strDebug[3])
   {
-    m_strDebug[3] = info4;
+    m_strDebug[3] = info.vsync;
     if (m_overlay[3])
       m_overlay[3]->Release();
     m_overlay[3] = new CDVDOverlayText();
     m_overlay[3]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[3]));
   }
+  if ("" != m_strDebug[4])
+  {
+    m_strDebug[4] = "";
+    if (m_overlay[4])
+      m_overlay[4]->Release();
+    m_overlay[4] = new CDVDOverlayText();
+    m_overlay[4]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[4]));
+  }
+  if ("" != m_strDebug[5])
```

Comment:
same thing here?

Reply:
You are right and this code can be removed:

```c++
  if ("" != m_strDebug[4])
  {
    m_strDebug[4] = "";
    if (m_overlay[4])
      m_overlay[4]->Release();
    m_overlay[4] = new CDVDOverlayText();
    m_overlay[4]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[4]));
  }
  if ("" != m_strDebug[5])
  {
    m_strDebug[5] = "";
    if (m_overlay[5])
      m_overlay[5]->Release();
    m_overlay[5] = new CDVDOverlayText();
    m_overlay[5]->AddElement(new CDVDOverlayText::CElementText(m_strDebug[5]));
  } 
```

And change `for` to ` i < 4` as debug info player only has 4 lines (is not need render empty lines).